src/flashbar/collapsible-flashbar.tsx

```diff
@@ -0,0 +1,392 @@
+                    key={getItemId(item)}
+                    ref={shouldUseStandardAnimation(item, index) ? transitionRootElement : undefined}
+                    transitionState={shouldUseStandardAnimation(item, index) ? state : undefined}
+                    {...item}
+                  />
+                )}
+              </li>
+            )}
+          </Transition>
+        ))}
+      </TransitionGroup>
+    </ul>
+  );
+
+  return (
+    <div
+      {...baseProps}
+      className={clsx(
+        baseProps.className,
+        styles.flashbar,
+        styles[`breakpoint-${breakpoint}`],
+        styles.stack,
+        isFlashbarStackExpanded && styles.expanded,
+        getVisualContextClassname('flashbar')
+      )}
+      ref={mergedRef}
+    >
+      <>
+        {isFlashbarStackExpanded && renderList()}
+        {isCollapsible && (
+          <div
+            className={clsx(
+              styles.toggle,
+              isVisualRefresh && styles['visual-refresh'],
+              isFlashbarStackExpanded ? styles.expanded : styles.collapsed,
+              transitioning && styles['animation-running']
+            )}
+            onClick={toggleCollapseExpand}
+            ref={toggleElementRef}
+          >
+            <span className={styles.status} role="status">
+              {toggleButtonText && <h2 className={styles.text}>{toggleButtonText}</h2>}
+              <span className={styles['types-count']} id={itemCountElementId}>
+                <LiveRegion>
+                  {counterTypes
+                    .map(
+                      ({ type, labelName }: { type: FlashType; labelName: LabelName }) =>
+                        `${countByType[type]} ${(i18nStrings && i18nStrings[labelName]) ?? ''}`
+                    )
+                    .join(', ')}
+                </LiveRegion>
+                <span aria-hidden="true" className={styles['types-count-visible']}>
+                  {counterTypes.map(({ type, labelName, iconName }) => (
+                    <NotificationTypeCount
+                      key={type}
+                      iconName={iconName}
+                      label={i18nStrings ? i18nStrings[labelName] : undefined}
+                      count={countByType[type]}
+                    />
+                  ))}
+                </span>
+              </span>
+            </span>
+            <button
+              aria-controls={flashbarElementId}
+              aria-describedby={itemCountElementId}
+              aria-expanded={isFlashbarStackExpanded}
+              aria-label={i18nStrings?.toggleButtonAriaLabel}
+              className={clsx(styles.button, isFlashbarStackExpanded && styles.expanded)}
+              {...isFocusVisible}
+            >
+              <InternalIcon className={styles.icon} size="normal" name="angle-down" />
+            </button>
+          </div>
+        )}
+        {!isFlashbarStackExpanded && renderList()}
+      </>
+    </div>
+  );
+}
+
+type LabelName =
```

Comment:
I would consider moving this code below into separate files.

Reply:
Done.